### ISSUE TITLE
Can't find AJAX handler

### ISSUE THREAD

**Issue by rickgoemans** [skydiver/october-plugin-forms#57]
The plugin seems promising but I can't get it to work.
I installed the plugin and added the emptyForm component to my partial (sections/contact.htm and the component's alias is emptyForm).
Then I added the following HTML snippet to my markup:
```
<form data-request="{{ __SELF__ }}::onFormSubmit">
	{{ form_token() }}
	<div id="{{ __SELF__ }}_forms_flash"></div>
	<!-- YOUR FORM FIELDS -->
	<input type="text" id="name" name="name" placeholder="uw naam" required="required">
	<div data-validate-for="name"></div>
	
	<input type="text" id="email" name="email" placeholder="uw e-mailadres" required="required">
	<div data-validate-for="email"></div>
	
	<textarea name="message" cols="30" rows="8" placeholder="uw bericht" required="required"></textarea>
	<div data-validate-for="message"></div>
	<!-- SUBMIT BUTTON -->
	<input type="submit" class="btn btn-primary" value="verzend">
</form>
```

When I submit the form the page just puts the parameters as GET in the URL bar. If I change 
`{{ __SELF__ }}`
to
`{{ emptyForm }}`
as suggested on the [reviews page of octoberCMS](http://octobercms.com/plugin/martin-forms#reviews) I get the following error:
AJAX handler 'emptyForm::onFormSubmit' was not found.

Am I missing something?

**Comment by skydiver** [comment]
Hello @rickgoemans, a few questions:

1) are you using jquery?
2) same problem using "Generic AJAX Form" component?
3) this problem still happens on fresh october install?

**Comment by rickgoemans** [comment]
1, Here's my script tag that's in my layout file(coming from the bootstrap3sass theme) :
```
<script src="{{ [
	'assets/vendor/jquery/dist/jquery.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/transition.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/alert.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/button.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/carousel.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/collapse.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/dropdown.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/modal.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/tooltip.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/popover.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/scrollspy.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/tab.js',
	'assets/vendor/bootstrap-sass/assets/javascripts/bootstrap/affix.js',
	'@framework',
	'@framework.extras',
	'assets/javascript/site.js'
] | theme }}"></script>
{% scripts %}
```

And I can access a AJAX handler on another page using "Data attributes API" and "Javascript API" so jQuery should work fine.

2. I skipped the "Generic AJAX Form" at first because this triggers a "partial not found error" for the recaptcha part. Even when I delete the recaptcha part it shows me the error:
`AJAX handler 'genericForm::onFormSubmit' was not found.`

3. I tried a fresh installation of OctoberCMS and it works(only the emptyForm tho, the Generic form stll shows the "partial not found error" for the recaptcha. I'm wondering why it isn't working with my project since I tried reinstalling the plugins I'm using in my fresh install and it's still working.

Any basic suggestions that might cause the problem?

**Comment by rickgoemans** [comment]
I found the reason behind the error.

I try to load the component within a partial. And apparently your plugin only checks the components of the page.

You can try to reproduce it yourself by doing this:
1. Make a page called (example /contact) and place the empty formcomponent on that page called "contactForm". It will work just fine.
2. Move the form code from the page you just created to a partial called "contact" and load that partial into you contact page. If you refer to data-request="{{ contactForm }}:onFormSubmit inside the partial it still works since you didn't remove the component on the page.
3. Move the component on the page to the partial. The error get triggered.

In short, the plugin doesn't fully support partials since it only checks for components on the page. A dirty solution would be added the component on each page where the partial with that form is loaded, but this requires a lot of editing if a single change is made. Another dirty solution might be adding the component to the layout file but I haven't tested it yet.

**Comment by skydiver** [comment]
There is some limitation when using components from partials.

Check this comment https://github.com/octobercms/october/issues/750#issuecomment-64059616

**Comment by kiretutu** [comment]
Talking about plugin magic something :??
For form ajax ?
If yes, someone tell me how use him...
